### ISSUE TITLE
Feature: Video Clipping

### ISSUE THREAD

**Issue by lafemmephile** [lafemmephile/core#5]
A class or function (possibly a *generator*) that can take a list of *videos*, a clip length value, and an optional filter function and return video clips. For example:
```python
def default_clipper(video_path: str, clip_length: int = 10) -> None:
    """Simplest clipping strategy."""
    # get vid object
    vid_obj = Video(video_path)

    # check vid length
    assert len(vid_obj) >= clip_length
    pass

def filter_clips(videos: List[str], filter_func: Callable[..., None] = default_clipper, *args) -> None:
    """Apply custom filter to determine where to clip videos."""
    # loop over videos
    for vid in videos:
        # filter out clips
        filter_func(vid, *args)
```

**Comment by lafemmephile** [comment]
CLI implementation possibilities:
+ `--length/-l` for the length of the clip in seconds (defaults to `10s`)
    ```console 
    femmephile clip -l 200 /path/to/video.mp4
    ```
+ `--out-dir/-o` for the output directory for the clips (defaults to `video_name_clips`)
    ```console
    femmephile clip -o /path/to/output/dir /path/to/video.mp4
    ```
+ `--filter/-f` for filtering clips based on content (defaults to `default_clipper`)
    ```console
    femmephile clip -f nsfw_filter /path/to/video.mp4
    ```

**Comment by lafemmephile** [comment]
Implemented with: 814eb12e474366b96995a2c299bd5c58a4859ba5
